pandas/core/nanops.py

```diff
@@ -864,6 +866,12 @@ def reduction(
                 result.fill(np.nan)
             except (AttributeError, TypeError, ValueError):
                 result = np.nan
+        elif is_object_dtype(dtype) and values.ndim == 1 and na_mask.any():
```

Comment:
do you have a test case that fails on non ndim==1?

Reply:
Yes, was getting a couple test failures otherwise, I think for reductions when the entire `DataFrame` has object dtype (I can't recall which tests exactly). I figured the subsetting `values[~mask]` is only going to make sense if `values` has one dimension.